apps/web/scripts/cleanupOlderPageVersions.ts

```
@@ -0,0 +1,114 @@
+}
+
+async function traverseBlocks(blocks: any[], callback: (block: any) => void) {
+  return await Promise.all(
+    blocks.map(async (block: any) => {
+      if (block.content) {
+        await traverseBlocks(block.content, callback);
+      }
+      await callback(block);
+    })
+  );
+}
+
+(async () => {
+  const { data: latestPublishedPages } = await supabase
+    .from("latest_page_publication")
+    .select("id, ipfs_cid")
+    .order("created_at", { ascending: false });
+
+  if (!latestPublishedPages) {
+    throw new Error("Failed to fetch latest published pages");
+  }
+
+  const usedCids = new Set<string>();
+
+  for (const { ipfs_cid } of latestPublishedPages) {
+    const page = await fetchPage(ipfs_cid);
+
+    await traverseBlocks(
+      page.data,
+      async (block: any) => {
+        if (block?.type === "ipfs-image") {
+          usedCids.add(block!.attrs!.cid);
+        }
+      }
+    )
+  }
+
+  const latestPublishedPageIds = latestPublishedPages.map(({ id }) => id);
+
+  const { data: oldPublishedPages } = await supabase
+    .from("page_publication")
+    .select("id, ipfs_cid, page_id")
+    .not("id", "in", `(${latestPublishedPageIds.join(",")})`);
+
+  if (!oldPublishedPages) {
+    throw new Error("Failed to fetch old published pages");
+  }
+
+  const unpinnedCids = new Set<string>();
+
+  for (const { id, ipfs_cid, page_id } of oldPublishedPages) {
+    const page = await fetchPage(ipfs_cid);
+    
+    await traverseBlocks(
+      page.data,
+      async (block: any) => {
+        if (block?.type === "ipfs-image") {
+          const { cid } = block!.attrs;
+          if (!usedCids.has(cid) && !unpinnedCids.has(cid)) {
+            console.log(`Unpinning file with CID ${cid} for page ${page_id}`);
+            await pinata.unpin(cid);
+          }
+          unpinnedCids.add(cid);
+        }
+      }
+    );
+
+    console.log(`Unpinning & deleting page with CID ${ipfs_cid} and page ID ${page_id}`);
+
+    await Promise.allSettled([
```

Do we want to log the result from this in case only one succeeds?
```ts
const [unpinResult, deleteResult] = await Promise.allSettled(...)
```

Or do we want to just use `Promise.all()` and let it throw if any of them fails? Either way we probably don't want it to be silen